src/lib/matrix/test/MatrixAssignmentTest.cpp

```diff
@@ -49,32 +80,32 @@ int main()
 
 	for (size_t i = 0; i < 3; i++) {
 		for (size_t j = 0; j < 3; j++) {
-			TEST(fabs(data[i * 3 + j] - m2(i, j)) < FLT_EPSILON);
+			EXPECT_FLOAT_EQ(data[i * 3 + j], m2(i, j));
 		}
 	}
 
-	TEST(!m2.isAllNan());
+	EXPECT_FALSE(m2.isAllNan());
 
 	float data_times_2[9] = {2, 4, 6, 8, 10, 12, 14, 16, 18};
 	Matrix3f m3(data_times_2);
 
-	TEST(isEqual(m, m2));
-	TEST(!(isEqual(m, m3)));
+	EXPECT_EQ(m, m2);
+	EXPECT_NE(m, m3);
 
 	m2 *= 2;
-	TEST(isEqual(m2, m3));
+	EXPECT_EQ(m2, m3);
 
 	m2 /= 2;
 	m2 -= 1;
 	float data_minus_1[9] = {0, 1, 2, 3, 4, 5, 6, 7, 8};
-	TEST(isEqual(Matrix3f(data_minus_1), m2));
+	EXPECT_EQ(Matrix3f(data_minus_1), m2);
 
 	m2 += 1;
-	TEST(isEqual(Matrix3f(data), m2));
+	EXPECT_EQ(Matrix3f(data), m2);
 
 	m3 -= m2;
 
-	TEST(isEqual(m3, m2));
+	EXPECT_EQ(m3, m2);
```

Comment:
No, because it's comparing two matrices and not two floats. So we use `EXPECT_EQ()` for comparing two variables/objects which for two matrices will use:
https://github.com/PX4/PX4-Autopilot/blob/6d0b11ca990ec1cdad2189a84d61bc5a9149b408/src/lib/matrix/matrix/Matrix.hpp#L341-L344

Reply:
ah i see now - this is matrix internal function. sorry - not really familiar with this library.. thought isequal was hold over from previous custom testing framework on matrix lib.